### ISSUE TITLE
Code stuck around loss.backward()  for forever

### ISSUE THREAD

**Issue by nileshkulkarni** [facebookresearch/clevr-iep#29]
Hi, 
I was using the code with the following configuration
`python scripts/train_model.py   --model_type PG   --num_train_samples 18000   --num_iterations 20000   --checkpoint_every 1000   --checkpoint_path data/program_generator.pt --batch_size 2`  on  a Tesla K40M. I followed the previous steps on data creation and all the data is available on the local machine.
I observed that the code is stuck in computing loss.backward on following [line](https://github.com/facebookresearch/clevr-iep/blob/21723998d470b72bd9796ffa5fefb96449424d38/scripts/train_model.py#L222). 
My Memory usage on the GPU stays at 296MB but the volatile gpu usage utilization is 0. Does anyone have any possible ideas on why this might be happening?

**Comment by nileshkulkarni** [comment]
The issues occurs if you use the pytorch 0.2.0 in a conda environment. Running the code with pytorch=0.1.12 does not have the same issue and runs the code.

**Comment by nileshkulkarni** [comment]
The issue occurs in pytorch has some relation to do with https://github.com/pytorch/pytorch/pull/3474
The worker nodes are in a deadlock and I guess it is a torch issue than a issue to this particular code base
